arrays/arrays.c

```diff
@@ -35,8 +40,15 @@ Array *create_array (int capacity) {
 void destroy_array(Array *arr) {
 
   // Free all elements
+  for(int i = 0; i < arr->count; i++){ // ensures deep deletion of all elements in the array
+    arr->elements[i] = NULL; 
```

Comment:
Awesome job here, so many people forget this for loop. Only thing is that you don't need to set the elements to NULL before freeing them. It's not a problem but just not strictly necessary.